src/main/java/com/atlassian/bitbucket/jenkins/internal/model/deployment/BitbucketDeploymentEnvironmentType.java

```diff
@@ -29,13 +26,16 @@
         this.weight = weight;
     }
 
-    public static Optional<BitbucketDeploymentEnvironmentType> fromName(@CheckForNull String name) {
+    @CheckForNull
+    public static BitbucketDeploymentEnvironmentType fromName(@CheckForNull String name) {
         if (StringUtils.isBlank(name)) {
-            return empty();
+            return null;
+        }
+        try {
+            return BitbucketDeploymentEnvironmentType.valueOf(name.toUpperCase(Locale.ROOT));
+        } catch (IllegalArgumentException e) {
+            return null;
```

Comment:
I changed to returning null, and also avoided the streaming of entries. This is called as a result of a form-post where the value comes from a dropdown, thus I think it is exceptional if it is not null nor a valid value. 

I opted for `null` instead of an `optional` as the usage of this was `fromName(..).orElse(null)` so you just wrapped a (essentially a)`null` and then unwrapped it to a `null` as soon as it was returned. I like optionals, but this didn't feel like the right place for using them given how this was actually used.

Reply:
I don't have a strong preference between `null` and `Optional` here. To me, `Optional` return types are nicer to use than nullable types because it forces you to handle the case where you can't coerce it into the right type.
It's not just used as `orElse(null)`, it is also used in validation:
```java
return BitbucketDeploymentEnvironmentType.fromName(environmentType)
        .map(validType -> FORM_VALIDATION_OK)
        .orElseGet(() -> FormValidation.error(Messages.DeployedToEnvironmentNotifierStep_EnvironmentTypeInvalid()));
```

But regardless, if we switch to `nulll` that's fine, but I would prefer that we didn't use exceptions to control flow. Maybe we can do something like this:
```java
for (BitbucketDeploymentEnvironmentType type : values()) {
    if (type.name().equalsIgnoreCase(name)) {
        return type;
    }
}
return null;
```

or something like this:
```java
    private static final Map<String, BitbucketDeploymentEnvironmentType> types =
            Maps.newHashMapWithExpectedSize(values().length);
    static {
        for (BitbucketDeploymentEnvironmentType type : values()) {
            types.put(type.name(), type);
        }
    }

  @CheckForNull
  public static BitbucketDeploymentEnvironmentType fromName(@CheckForNull String name) {
    return types.get(name);
  }
```

Reply:
> To me, Optional return types are nicer to use than nullable types because it forces you to handle the case where you can't coerce it into the right type.

Totally agree, however, what we did was exactly one thing which was `..Type.fromName(stripToNull(environmentType)).orElse(null)` so you take this and turn it straight to null, not doing anything more. 
I don't feel strongly about it, but it just feels like you essentially wrap the value and then unwrap it right away, so the handling is just more code without any benefit.

Reply:
In _one_ usage we do that, yes. In another, we don't. Currently, it's only being used in two instances, but we don't know what future usages are going to do. Being a model class it feels like we should be using `Optional`